subgraph/src/listeners.ts

```diff
@@ -0,0 +1,80 @@
+import { Bytes } from '@graphprotocol/graph-ts'
+import {ERC20Mock as ERC20Contract} from '../types/ERC20/ERC20Mock'
+import { OfferCreated as OfferCreatedEvent } from '../types/OTC/OTC'
+import { Offer as OfferContract, OfferAccepted as OfferAcceptedEvent } from '../types/Offer/Offer'
+import { Offer } from '../types/schema'
+import { generateEvent, getOrCreateAsset, getAssetByAddress } from './utils'
+
+
+const getPrice = (address: string): i32 => {
+  const asset = getAssetByAddress(address)
+  const rate = asset.rate
+
+  if (!rate.startsWith('0x')) {
+    return i32(parseInt(rate))
+  }
+
+  return 0
+}
+
+export function handleOfferCreated(event: OfferCreatedEvent): void {
+  const sourceAssetAddress = event.params.srcAsset.toHexString()
+  const destAssetAddress = event.params.destAsset.toHexString()
+  const sourceAsset = getOrCreateAsset(sourceAssetAddress)
+  const destAsset = getOrCreateAsset(destAssetAddress)
+  const sourceAssetContract = ERC20Contract.bind(event.params.srcAsset)
+  const destAssetContract = ERC20Contract.bind(event.params.destAsset)
+  const offerContract = OfferContract.bind(event.params.offerAddress)
+
+  sourceAsset.decimals = sourceAssetContract.decimals()
+  destAsset.decimals = destAssetContract.decimals()
+  
+  sourceAsset.save()
```

Comment:
Not sure about saving the entity for every event call, might create unnecessary DB congestion. If it already exists, save can be avoided

Reply:
Nice catch. I updated.

Reply:
Resolved in https://github.com/polymorpher/otc-country/pull/23